### PR TITLE
iter_size parameter

### PR DESCRIPTION
**Pull request by Kipok**
For now only supports Horovod mode and accumulates gradients in fp32, instead of fp16 (since for that we would need to move MPOptimizer logic into optimize_loss function).

### CODE REVIEW

File: open_seq2seq/models/model.py [hunk 1]
```diff
@@ -168,6 +169,8 @@ class docs.
     * **summaries** (list) --- which summaries to log. Could contain
       "learning_rate", "gradients", "gradient_norm", "global_gradient_norm",
       "variables", "variable_norm".
+    * **iter_size** (int) --- same as in nvcaffe, the gradients will be
```
Comment (okuchaiev) [inline open_seq2seq/models/model.py]: lets remove nvcaffe reference: 1) it is not technically same due to fp32 aggregation and 2) same as in bla not a very good style for documentation

Comment (okuchaiev) [reply open_seq2seq/models/model.py]: Maybe add more about the fact that it can be used to "simulate" larger batches

File: open_seq2seq/optimizers/optimizers.py [hunk 2]
```diff
@@ -72,69 +72,26 @@ def get_regularization_loss(scope=None, name="total_regularization_loss"):
...
     else:
-      return gradients
-
-  def apply_gradients(self, grads_and_vars, global_step=None, name=None):
-    """Calls this same method on the underlying optimizer."""
-    return self._optimizer.apply_gradients(grads_and_vars, global_step, name)
+      return grads_and_vars
+  else:
+    # TODO: implement this
+    pass
```
Comment (okuchaiev) [inline open_seq2seq/optimizers/optimizers.py]: please raise exception here instead of pass

**Comment by okuchaiev** [comment]
maybe rename "iter_size" to something along the lines of "virtual_batch_scale_factor"?
